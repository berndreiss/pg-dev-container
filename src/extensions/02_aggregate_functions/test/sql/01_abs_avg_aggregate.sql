-- Direct function call
SELECT abs_avg(1234);
SELECT abs_avg(0);
SELECT abs_avg(-123);
SELECT abs_avg(NULL);

-- Test using generate series
SELECT abs_avg(i) FROM generate_series(1, 10) i;
SELECT abs_avg(i) FROM generate_series(1, 100) i;
SELECT abs_avg(i) FROM generate_series(-10, 100) i;
SELECT abs_avg(i) FROM generate_series(-100, 100) i;

-- Test using a table
CREATE TABLE test_table AS SELECT i FROM generate_series(1, 100) i;
SELECT abs_avg(i) FROM test_table;
DROP TABLE test_table;
