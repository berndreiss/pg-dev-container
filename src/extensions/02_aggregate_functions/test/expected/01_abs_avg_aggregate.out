-- Direct function call
SELECT abs_avg(1234);
 abs_avg 
---------
    1234
(1 row)

SELECT abs_avg(0);
 abs_avg 
---------
       0
(1 row)

SELECT abs_avg(-123);
 abs_avg 
---------
     123
(1 row)

SELECT abs_avg(NULL);
 abs_avg 
---------
        
(1 row)

-- Test using generate series
SELECT abs_avg(i) FROM generate_series(1, 10) i;
 abs_avg 
---------
     5.5
(1 row)

SELECT abs_avg(i) FROM generate_series(1, 100) i;
 abs_avg 
---------
    50.5
(1 row)

SELECT abs_avg(i) FROM generate_series(-10, 100) i;
      abs_avg       
--------------------
 45.990990990990994
(1 row)

SELECT abs_avg(i) FROM generate_series(-100, 100) i;
      abs_avg      
-------------------
 50.24875621890547
(1 row)

-- Test using a table
CREATE TABLE test_table AS SELECT i FROM generate_series(1, 100) i;
SELECT abs_avg(i) FROM test_table;
 abs_avg 
---------
    50.5
(1 row)

DROP TABLE test_table;
